p_sets/p_set_1.rb

```diff
@@ -0,0 +1,91 @@
+# http://cdn.cs50.net/2016/x/psets/1/pset1/pset1.html
+# SMART WATER
+BOTTLE_VOLUME = 16
+WATER_FLOW_RATE = 192
+
+def bottle_reporter
+  puts "That's #{bottle_calculator ask_minutes} bottles of water"
+end
+
+def bottle_calculator(minutes)
+  minutes * WATER_FLOW_RATE / BOTTLE_VOLUME
+end
+
+def ask_minutes
+  puts "How many mins do you shower for?"
+  minutes = gets.chomp.to_i
+end
+
+# ITSA MARIO
+SPACE =      " "
+BRICKS =     "#"
+APEX_WIDTH = 2
+MIN_HEIGHT = 0
+MAX_HEIGHT = 23
+
+def ask_height
+  while true
+    puts "height?"
+    height = gets.chomp.to_i
+    break if valid_input?(height)
+  end
+  height
+end
+
+def valid_input?(height)
+  height <= MAX_HEIGHT && height > MIN_HEIGHT
+end
+
+def calculate_row_length(height)
+  row_length = height + 1
+end
+
+def pyramid_printer(row_length)
+  n = APEX_WIDTH
+  for n in n..row_length
```

Comment:
Idiomatic Ruby tends to avoid using `for` loops directly. There's nearly always a better iterator, like

```
(APEX_WIDTH..row_length).each do |bricks_in_row|
```
or even better IMO:
```
APEX_WIDTH.upto(row_length) do |bricks_in_row|
```

Ruby prefers using iterators and blocks because they can be passed around as objects: it's possible to refactor the contents of the `do…end` block and pass it in as an argument, for example. The end result is the same as a bare `for` loop and there's a lot more flexibility, so it's a better habit to get into